flexget/plugins/input/imdb_watchlist.py

```diff
@@ -84,72 +85,144 @@ def on_task_input(self, task, config):
             url = 'http://www.imdb.com/user/%s/%s' % (config['user_id'], config['list'])
+            if config['list'] == 'watchlist':
+                params = {'view': 'detail'}
         else:
             url = 'http://www.imdb.com/list/%s' % config['list']
+        if 'all' not in config['type']:
+            title_types = []
+            for title_type in config['type']:
+                title_types.append(TITLE_TYPE_MAP[title_type])
+            params['title_type'] = ','.join(title_types)
+            params['sort'] = 'list_order%2Casc'
+
+        if config['list'] == 'watchlist':
+            entries = self.parse_react_widget(task, config, url, params, headers)
+        else:
+            entries = self.parse_html_list(task, config, url, params, headers)
+        return entries
 
-        headers = {'Accept-Language': config.get('force_language')}
+    def fetch_page(self, task, url, params, headers):
         log.debug('Requesting: %s %s', url, headers)
-
         try:
             page = task.requests.get(url, params=params, headers=headers)
         except HTTPError as e:
             raise plugin.PluginError(e.args[0])
         if page.status_code != 200:
             raise plugin.PluginError('Unable to get imdb list. Either list is private or does not exist.')
+        return page
 
-        soup = get_soup(page.text)
+    def parse_react_widget(self, task, config, url, params, headers):
+        page = self.fetch_page(task, url, params, headers)
+        try:
+            re_result = re.search('IMDbReactInitialState.push\((.+?)\);\n', page.text)
+            json_text = re_result.group(1)
+            json_vars = json.loads(json_text)
+        except (TypeError, AttributeError, ValueError) as e:
+            raise plugin.PluginError(
+                'Unable to get imdb list from imdb react widget.' +
+                ' Either the list is empty or the imdb parser plugin is broken.' +
+                ' Original error: %s.' % e.args[0])
+        total_item_count = 0
+        if 'list' in json_vars and 'items' in json_vars['list']:
+            total_item_count = len(json_vars['list']['items'])
+        if not total_item_count:
+            log.verbose('No movies were found in imdb list: %s', config['list'])
+            return
+        imdb_ids = []
+        for item in json_vars['list']['items']:
+            if is_valid_imdb_title_id(item['const']):
+                imdb_ids.append(item['const'])
+        params = {'ids': ','.join(imdb_ids)}
+        url = 'http://www.imdb.com/title/data'
+        try:
+            page = task.requests.get(url, params=params)
+        except HTTPError as e:
+            raise plugin.PluginError(e.args[0])
+        if page.status_code != 200:
+            raise plugin.PluginError('Unable to get imdb title data. Html status code was: %d.' % page.status_code)
+        json_data = json.loads(page.text)
+        log.verbose('imdb list contains %d items' % len(json_data))
```

Comment:
All logging calls should use implicit string interpolation ie. the logger takes an arbitrary number of arguments that it will use for string interpolation. As an example, the above line should be `log.verbose('imdb list contains %d items', len(json_data))` (notice the comma).